metricbeat/mb/testing/modules.go

```diff
@@ -217,29 +217,34 @@ func NewPushMetricSetV2(t testing.TB, config interface{}) mb.PushMetricSetV2 {
 	return pushMetricSet
 }
 
-type capturingReporterV2 struct {
-	events []mb.Event
-	done   chan struct{}
+// CapturingReporterV2 stores all the events and errors from a metricset's
+// Run method.
+type CapturingReporterV2 struct {
+	Events []mb.Event
+	DoneC  chan struct{}
 }
 
-func (r *capturingReporterV2) Event(event mb.Event) bool {
-	r.events = append(r.events, event)
+// Event stores the passed-in event into the events array
+func (r *CapturingReporterV2) Event(event mb.Event) bool {
+	r.Events = append(r.Events, event)
 	return true
 }
 
-func (r *capturingReporterV2) Error(err error) bool {
-	r.events = append(r.events, mb.Event{Error: err})
+// Error stores the given error into the errors array.
+func (r *CapturingReporterV2) Error(err error) bool {
+	r.Events = append(r.Events, mb.Event{Error: err})
 	return true
 }
 
-func (r *capturingReporterV2) Done() <-chan struct{} {
-	return r.done
+// Done returns the Done channel for this reporter.
+func (r *CapturingReporterV2) Done() <-chan struct{} {
+	return r.DoneC
 }
 
 // RunPushMetricSetV2 run the given push metricset for the specific amount of
 // time and returns all of the events and errors that occur during that period.
 func RunPushMetricSetV2(duration time.Duration, metricSet mb.PushMetricSetV2) []mb.Event {
```

Comment:
I think it would make sense to add another parameter (`maxEvents`) to RunPushMetricSetV2 for specifying the maximum number of events to wait for. So then the behavior would be run until you reach `duration` (probably should rename to timeout) or until the maximum number of events is received (if `maxEvents > 0`).